packages/scm/src/browser/history/scm-history-widget.tsx

```diff
@@ -62,36 +92,89 @@ export class GitHistoryWidget extends GitNavigableListWidget<GitHistoryListNode>
         state: 'loading',
     } | {
         state: 'ready',
-        commits: GitCommitNode[];
+        commits: ScmCommitNode[];
     } | {
         state: 'error',
         errorMessage: React.ReactNode
     };
 
+    protected readonly toDisposeOnRepositoryChange = new DisposableCollection();
+
+    protected historySupport: ScmHistorySupport | undefined;
+
     constructor(
+        @inject(ScmService) protected readonly scmService: ScmService,
         @inject(OpenerService) protected readonly openerService: OpenerService,
-        @inject(GitCommitDetailOpenHandler) protected readonly detailOpenHandler: GitCommitDetailOpenHandler,
         @inject(ApplicationShell) protected readonly shell: ApplicationShell,
         @inject(FileSystem) protected readonly fileSystem: FileSystem,
-        @inject(Git) protected readonly git: Git,
         @inject(ScmAvatarService) protected readonly avatarService: ScmAvatarService,
         @inject(WidgetManager) protected readonly widgetManager: WidgetManager,
-        @inject(GitDiffContribution) protected readonly diffContribution: GitDiffContribution) {
+    ) {
         super();
-        this.id = GIT_HISTORY_ID;
+        this.id = SCM_HISTORY_ID;
         this.scrollContainer = 'git-history-list-container';
-        this.title.label = GIT_HISTORY_LABEL;
-        this.title.caption = GIT_HISTORY_LABEL;
-        this.title.iconClass = 'fa git-history-tab-icon';
+        this.title.label = SCM_HISTORY_LABEL;
+        this.title.caption = SCM_HISTORY_LABEL;
+        this.title.iconClass = 'fa scm-history-tab-icon';
         this.title.closable = true;
         this.addClass('theia-git');
         this.resetState();
         this.cancelIndicator = new CancellationTokenSource();
     }
 
+    @postConstruct()
+    protected init(): void {
+        this.refreshOnRepositoryChange();
+        this.toDispose.push(this.scmService.onDidChangeSelectedRepository(() => this.refreshOnRepositoryChange()));
+    }
+
+    protected refreshOnRepositoryChange() {
+        this.toDisposeOnRepositoryChange.dispose();
+
+        const repository = this.scmService.selectedRepository;
+        if (repository) {
+            this.historySupport = repository.input.get<ScmHistorySupport>(ScmHistorySupport);
```

Comment:
Is there a cleaner way to get the `ScmHistorySupport`?

Reply:
Yes, there is a cleaner way, we could use the same way the optional AmendSupport implementation is obtained.  However this is more flexible and allows stuff to be rebound using Inversify.